crates/assists/src/utils.rs

```diff
@@ -98,13 +98,14 @@ pub fn filter_assoc_items(
 
     items
         .iter()
-        .map(|i| {
-            #[allow(deprecated)]
-            match i {
-                hir::AssocItem::Function(i) => ast::AssocItem::Fn(i.source_old(db).value),
-                hir::AssocItem::TypeAlias(i) => ast::AssocItem::TypeAlias(i.source_old(db).value),
-                hir::AssocItem::Const(i) => ast::AssocItem::Const(i.source_old(db).value),
-            }
+        // Note: This throws away items with no source.
```

Comment:
This is an interesting case! In the future(not this PR), we should add hir -> ast function which either uses an existing source, or just renders the hir directly.

Reply:
Yeah. I've seen reference to using hir for more in a few comments in the codebase.